### ISSUE TITLE
Pull files endpoint errors with 504 on large files and doesn't log the activity

### ISSUE THREAD

**Issue by LindaJuffermans** [pterodactyl/panel#4730]
### Current Behavior

`/servers/{server}/files/pull` responds with 504 in case the download takes too long:

```
{
    "errors": [
        {
            "code": "DaemonConnectionException",
            "status": "504",
            "detail": "Could not establish a connection to the machine running this server. Please try again."
        }
    ]
}
```

The download meanwhile continues in the background.
Additionally, no activity is logged in /server/{server}/activity so there is no audit trail.

### Expected Behavior

With large downloads, you can't keep a connection for 30 minutes.
The API should just respond 204 indicating that it has started.
Additionally, the /activity/ page should show that the download has started.

(Ideally, the /activity/ page could show when the download has completed; this also provides feedback to the user when the task has completed. I will raise this as a feature request.)

### Steps to Reproduce

Post a download request via `/servers/{server}/files/pull` with `foreground: true` using a large file to be downloaded that takes longer than the API timeout.

### Panel Version

1.8.1

### Wings Version

1.6.4

### Games and/or Eggs Affected

N/A

### Docker Image

N/A

### Error Logs

```bash
{
    "errors": [
        {
            "code": "DaemonConnectionException",
            "status": "504",
            "detail": "Could not establish a connection to the machine running this server. Please try again."
        }
    ]
}
```

### Is there an existing issue for this?

- [X] I have searched the existing issues before opening this issue.
- [X] I have provided all relevant details, including the specific game and Docker images I am using if this issue is related to running a server.
- [X] I have checked in the Discord server and believe this is a bug with the software, and not a configuration issue with my specific system.

**Comment by devnote-dev** [comment]
> `/servers/{server}/files/pull` responds with 504 in case the download takes too long:
> [...]

Assuming this is the client API endpoint (`/api/client/servers/:id/files/pull`) this error would likely be caused by a current or previous pull request failing. Under normal circumstances you should only get a 204 response because it is asynchronous.

> Additionally, no activity is logged in /server/{server}/activity so there is no audit trail.

The Wings endpoint doesn't log it for some reason, it should be added though as well as an activity log for failed pull requests.

> With large downloads, you can't keep a connection for 30 minutes.

Assuming you're referring to the client API endpoint again, this isn't the case. For the Wings API endpoint you would only have this issue if you set the `foreground` option to true (as well as configured your HTTP client to not timeout on the request if necessary), so I don't really see this as an issue. It makes sense to use the `foreground` option if you know that the resource you are pulling is small and/or can be downloaded relatively quickly, but for larger/longer resources there is no benefit.

**Comment by LindaJuffermans** [comment]
> Assuming this is the client API endpoint (/api/client/servers/:id/files/pull) this error would likely be caused by a current or previous pull request failing. Under normal circumstances you should only get a 204 response because it is asynchronous.
Yes, it's the client API endpoint. It's the response of a new pull request which is actually successful; I can see the filesize increasing in the panel's filemanager.

> The Wings endpoint doesn't log it for some reason, it should be added though as well as an activity log for failed pull requests.
It didn't fail, the file is there.

> Assuming you're referring to the client API endpoint again, this isn't the case.
Yes, it was. And yes, I did set `foreground: true` (sorry, forgot to mention it). Since there is no completion feedback with `foreground: false`, I was hoping I would be able to see this complete.

> as well as configured your HTTP client to not timeout on the request if necessary
This wasn't a timeout on my Postman client, I believe it was a timeout on the client API on its call to the Wings API.
That's why I got a 504 response from the client API and not a timeout.

**Comment by devnote-dev** [comment]
Side note: with GitHub markdown you have to put two newlines after a quote otherwise it counts the next line as part of the quote. 😉 

> It didn't fail, the file is there.

Ah right. But in general, having an activity log for successful and failed pull requests would be useful.

> Yes, it was. And yes, I did set `foreground: true` (sorry, forgot to mention it). Since there is no completion feedback with `foreground: false`, I was hoping I would be able to see this complete.

If you use the Wings API endpoint and set `foreground` to false, it will return the download identifier which you can then poll/query using the [Get Server Files Pull](https://github.com/devnote-dev/ptero-notes/blob/main/wings/server_files.md#get-apiserversuuidfilespull) endpoint. It's not the nicest solution but it's the only available one until #4143 is implemented.

> This wasn't a timeout on my Postman client, I believe it was a timeout on the client API on its call to the Wings API.
That's why I got a 504 response from the client API and not a timeout.

Can you share the Wings logs for this please?
